### PR TITLE
Package petr4.0.1

### PR DESCRIPTION
**Pull request by jnfoster**
### `petr4.0.1`
Petr4: Formal Semantics for the P4 Programming Language

---
* Homepage: https://github.com/cornell-netlab/petr4
* Source repo: git+https://github.com/cornell-netlab/petr4/
* Bug tracker: https://github.com/cornell-netlab/petr4/issues

---
:camel: Pull-request generated by opam-publish v2.0.0

### CODE REVIEW

File: packages/petr4/petr4.0.1/opam [hunk 1]
```diff
@@ -0,0 +1,28 @@
+opam-version: "2.0"
+synopsis: "Petr4: Formal Semantics for the P4 Programming Language"
+maintainer: "jnfoster@cs.cornell.edu"
+authors: ["Nate Foster <jnfoster@cs.cornell.edu>"]
+homepage: "https://github.com/cornell-netlab/petr4"
+bug-reports: "https://github.com/cornell-netlab/petr4/issues"
+dev-repo: "git+https://github.com/cornell-netlab/petr4/"
+license: "Apache2"
+build: [
+  ["dune" "build" "--root" "." "-j" jobs "@install"]
```
Comment (kit-ty-kate) [inline packages/petr4/petr4.0.1/opam]:
```suggestion
  ["dune" "build" "-p" name "-j" jobs]
```
This is more the usual command. Using -p allows to only select the current package and not everything that could be inside the source code

File: packages/petr4/petr4.0.1/opam [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+]
+depends: [
+  "ocaml" { build & >= "4.05.0" }
```
Comment (kit-ty-kate) [inline packages/petr4/petr4.0.1/opam]:
```suggestion
  "ocaml" { >= "4.05.0" }
```
OCaml is not a build dependency in most cases.

**Comment by kit-ty-kate** [comment]
Compilation fails with:
```
#=== ERROR while compiling petr4.0.1 ==========================================#
# context              2.0.3 | linux/x86_64 | ocaml-base-compiler.4.07.1 | pinned(https://github.com/cornell-netlab/petr4/archive/0.1.tar.gz)
# path                 ~/.opam/4.07/.opam-switch/build/petr4.0.1
# command              ~/.opam/4.07/bin/dune build --root . -j 71 @install
# exit-code            1
# env-file             ~/.opam/log/petr4-20-9d8f57.env
# output-file          ~/.opam/log/petr4-20-9d8f57.out
### output ###
# File "bin/dune", line 4, characters 18-31:
# 4 |   (libraries core core_extended p4)
#                       ^^^^^^^^^^^^^
# Error: Library "core_extended" not found.
# Hint: try: dune external-lib-deps --missing -j 71 --root . @install
```

**Comment by mseri** [comment]
ping @jnfoster

File: packages/petr4/petr4.0.1/opam (hunk 1 above)
Comment (jnfoster) [reply packages/petr4/petr4.0.1/opam]: Thanks. We've replaced this with
```
build: [
  [make]
]
```
where the `Makefile` includes a default target:
```
build:
	dune build --profile release
```
Is that acceptable?

File: packages/petr4/petr4.0.1/opam (hunk 2 above)
Comment (jnfoster) [reply packages/petr4/petr4.0.1/opam]: Thanks. Our mistake.

**Comment by jnfoster** [comment]
FWIW, the build process seems to be slightly flaky. 

I started a run last night, and everything succeeded except for the MacOS Travis CI build, which failed with an error message saying that it could not fetch the archives for a large number of packages (unfortunately I neither saved the list nor the error message). 

I made a trivial change to the OPAM file to trigger another run this morning, and now it succeeds.

**Comment by jnfoster** [comment]
Also, is it normal for Camelus to be taking more than 3 hours to run?

File: packages/petr4/petr4.0.1/opam (hunk 1 above)
Comment (jnfoster) [reply packages/petr4/petr4.0.1/opam]: Thanks to some help from @smolkaj I've updated to a standard build declaration.

File: packages/petr4/petr4.0.1/opam [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+license: "Apache2"
+build: [
+  ["dune" "subst"] {pinned}
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
+  ["dune" "build" "@doc" "-p" name "-j" jobs] {with-doc}
+]
+depends: [
+  "ocaml" {>= "4.04.0"}
+  "dune"
```
Comment (kit-ty-kate) [inline packages/petr4/petr4.0.1/opam]:
```suggestion
  "dune" {build}
```
This dependency is only needed at build time

Comment (jnfoster) [reply packages/petr4/petr4.0.1/opam]: Thanks. Done.

**Comment by kit-ty-kate** [comment]
Thanks

**Pull request merged**